src/m365/entra/commands/group/group-set.ts

```diff
@@ -0,0 +1,292 @@
+            return `The maximum amount of characters for 'mailNickname' is 64.`;
+          }
+        }
+
+        if (args.options.ownerIds) {
+          const ids = args.options.ownerIds.split(',').map(i => i.trim());
+          if (!validation.isValidGuidArray(ids)) {
+            const invalidGuid = ids.find(id => !validation.isValidGuid(id));
+            return `'${invalidGuid}' is not a valid GUID for option 'ownerIds'.`;
+          }
+        }
+
+        if (args.options.ownerUserNames) {
+          const isValidUserPrincipalNameArray = validation.isValidUserPrincipalNameArray(args.options.ownerUserNames.split(',').map(u => u.trim()));
+          if (isValidUserPrincipalNameArray !== true) {
+            return `User principal name '${isValidUserPrincipalNameArray}' is invalid for option 'ownerUserNames'.`;
+          }
+        }
+
+        if (args.options.memberIds) {
+          const ids = args.options.memberIds.split(',').map(i => i.trim());
+          if (!validation.isValidGuidArray(ids)) {
+            const invalidGuid = ids.find(id => !validation.isValidGuid(id));
+            return `'${invalidGuid}' is not a valid GUID for option 'memberIds'.`;
+          }
+        }
+
+        if (args.options.memberUserNames) {
+          const isValidUserPrincipalNameArray = validation.isValidUserPrincipalNameArray(args.options.memberUserNames.split(',').map(u => u.trim()));
+          if (isValidUserPrincipalNameArray !== true) {
+            return `User principal name '${isValidUserPrincipalNameArray}' is invalid for option 'memberUserNames'.`;
+          }
+        }
+
+        if (args.options.visibility && ['Public', 'Private'].indexOf(args.options.visibility!) === -1) {
+          return `Option 'visibility' must be one of the following values: Public, Private.`;
+        }
+
+        return true;
+      }
+    );
+  }
+
+  #initOptionSets(): void {
+    this.optionSets.push(
+      { options: ['id', 'displayName'] }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    let groupId = args.options.id;
+    let ownerIds: string[] = [];
+    let memberIds: string[] = [];
+
+    try {
+      if (args.options.displayName) {
+        if (this.verbose) {
+          await logger.logToStderr(`Retrieving group id...`);
+        }
+
+        groupId = await entraGroup.getGroupIdByDisplayName(args.options.displayName);
+      }
+
+      const manifest = this.createRequestBody(args.options);
+      const requestOptions: CliRequestOptions = {
+        url: `${this.resource}/v1.0/groups/${groupId}`,
+        headers: {
+          accept: 'application/json;odata.metadata=none'
+        },
+        data: manifest
+      };
+
+      ownerIds = await this.getUserIds(logger, args.options.ownerIds, args.options.ownerUserNames);
+      memberIds = await this.getUserIds(logger, args.options.memberIds, args.options.memberUserNames);
+
+      await request.patch(requestOptions);
```

Comment:
Can't we patch the group before retrieving the user IDs? This splits the code nicely into two parts: updating group & setting members and owners.